### DISCUSSION TITLE
Visual Studio C/C++ compiler license

### DISCUSSION THREAD

**Discussion by rcotterill** [actions/virtual-environments#7233] [Installed software]
Hello,

Is there a license requirement to use Microsoft's C/C++ compiler in a GitHub Action?

We have a repo that uses `msbuild` for CI builds. The end customer uses Visual Studio, but we need to share the repo with a company that does not have Visual Studio licenses.

From the action in our repo:
`C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.34.31933\bin\HostX86\x86\CL.exe`

There is a similar question here https://github.com/actions/runner-images/discussions/3271 but it is about Visual Studio Enterprise not the C/C++ compiler tools.

Thank you,
Richard

**Comment by JulesSudolR** [comment]
Hello Richard,

According to Microsoft's licensing terms, the use of the Visual Studio Build Tools and the C/C++ compiler in GitHub Actions is allowed as long as the individual or organization using them has a valid Visual Studio license or is using the Community Edition of Visual Studio.

You mentioned that your end customer uses Visual Studio, so they likely have a valid license to use the build tools. However, if you are sharing the repo with a company that does not have Visual Studio licenses, they may need to obtain a license or use the Community Edition of Visual Studio in order to use the build tools legally.

It's always best to check with Microsoft or a legal expert for specific licensing questions, but I hope this information helps!
